armi/physics/neutronics/crossSectionSettings.py

```diff
@@ -125,65 +181,136 @@ def isPregenerated(self):
         """True if this points to a pre-generated XS file."""
         return self.fileLocation is not None
 
+    def _validate(self):
+        """
+        Perform validation checks on the set of inputs.
+
+        This makes some design assumptions about the lattice
+        physics code (i.e., critical buckling search is not
+        a valid option for non-0D problems), so if different
+        behavior is desired then it is recommended to subclass
+        this and to make a new cross section setting.
+
+        Notes
+        -----
+        This checks for any inconsistencies in the definition of
+        the inputs.
+        """
+        if self.fileLocation is None and self.geometry is None:
+            raise ValueError(f"{self} is missing a geometry input or a file location.")
+
+        if self.fileLocation is not None and self.geometry is not None:
+            raise ValueError(
+                f"The file location and geometry inputs cannot be combined in {self}. "
+                f"File location set to {self.fileLocation}. "
+                f"Geometry set to {self.geometry}"
+            )
+
+        # Check for valid inputs when the file location is supplied.
+        invalids = []
+        if self.fileLocation is not None:
+            for var, val in self:
+                # Skip the ``xsID`` and ``fileLocation`` attributes.
+                if var in [CONF_XSID, CONF_FILE_LOCATION, CONF_BLOCK_REPRESENTATION]:
+                    continue
+                if val is not None:
+                    invalids.append((var, val))
```

Comment:
Is there a clear reason why the voluptuous "extra keys" feature won't work for this?  That might be useful to document since we're kind of mixing vol validation with manual validation.

https://github.com/alecthomas/voluptuous#extra-dictionary-keys

Reply:
Not sure if extra keys would work. Seems like it would make the Schema very complex and probably more difficult to understand.

Reply:
See comment about. The Schema is actually only used for loading the dictionary before it gets loaded into the XSModelingOption class. Even if the Schema handled this, the XSModelingOption could still then be incorrectly set/defined.